### ISSUE TITLE
DONT USE LIBRARY BAD CODE

### ISSUE THREAD

**Issue by kampaidev** [valyala/fasthttp#1670]
JUST USE GIN OR ANY OTHER LIBRARY THIS WILL WASTE YOUR FUCKING TIME 

handle the buffers...

panic: runtime error: slice bounds out of range [:4914] with capacity 4096

**Comment by erikdubbelboer** [comment]
I agree! Only use fasthttp when you know what you are doing. People who don't know what they are doing will get these kinds of errors as they don't read the documentation and don't understand the the tradeoffs fasthttp makes for performance.

**Comment by kampaidev** [comment]
> I agree! Only use fasthttp when you know what you are doing. People who don't know what they are doing will get these kinds of errors as they don't read the documentation and don't understand the the tradeoffs fasthttp makes for performance.
> 
> It's even in the readme: https://github.com/valyala/fasthttp#fasthttp-might-not-be-for-you

I do know what I'm doing and with the amount of new fucking panics that I get from running millions of req a second every day it's some new shit with my users. I have to now write my whole fucking backend in gin, I really didnt think you guys were that breadead to spend that much time on a library just for it not to work

**Comment by kampaidev** [comment]
I'm using recover too btw still panics

**Comment by kampaidev** [comment]
I'm saving anyone else's time this library should be removed from git no joke, along with gofiber and any other library that uses this bullshit as a base I'd wipe my ass with it seriously

**Comment by kampaidev** [comment]
Made a whole damn library and can't even read buffers correctly

**Comment by erikdubbelboer** [comment]
Let me guess, you are keeping a reference to a `[]byte` returned be one of the fasthttp functions, even though the documentation clearly states:
```
The returned value is valid until the request is released, either though ReleaseRequest or your
request handler returning. Do not store references to returned value. Make copies instead. 
```

**Comment by kampaidev** [comment]
No I'm not lol. The buffer size was too slow, and lead to a crash even though that shouldn't have happened. I don't store any references to anything... Now I understand why the library doesn't work you make assumptions that are baseless without actually looking into and testing things.

This is actually the main problem I have with the library, no checks nothing. You've literally wasted your time making this I'm sorry for you

**Comment by kampaidev** [comment]
And TRUST ME ever error I've read the docs. All 20 of them. Look anybody reading this, I've scaled my apis to a point where 99% of people can't even come close to. Continue to say dumb shit like "you dont know what you're doing" because I really dont give a fuck my work speaks for itself. Fuck you for wsasting my time

**Comment by kampaidev** [comment]
Well made, tested library used by many big companies (https://github.com/gin-gonic/gin). Don't use this, honestly I don't care if you do, you'll just end up having to rewrite your whole backend once you scale regardless

**Comment by kampaidev** [comment]
I've said what I said. Recoding my backend in gin, like it was originally. Goodbye.
